OpenRA.Mods.Common/Activities/Air/HeliReturnToBase.cs

```diff
@@ -44,13 +44,28 @@ public override Activity Tick(Actor self)
 			{
 				var rearmBuildings = heli.Info.RearmBuildings;
 				var nearestHpad = self.World.ActorsHavingTrait<Reservable>()
-									.Where(a => a.Owner == self.Owner && rearmBuildings.Contains(a.Info.Name))
-									.ClosestTo(self);
+					.Where(a => a.Owner == self.Owner && rearmBuildings.Contains(a.Info.Name))
+					.ClosestTo(self);
 
 				if (nearestHpad == null)
 					return ActivityUtils.SequenceActivities(new Turn(self, initialFacing), new HeliLand(self, true), NextActivity);
 				else
-					return ActivityUtils.SequenceActivities(new HeliFly(self, Target.FromActor(nearestHpad)));
+				{
+					var distanceFromHelipad = (nearestHpad.CenterPosition - self.CenterPosition).HorizontalLength;
+					var distanceLength = heli.Info.WaitDistanceFromResupplyBase.Length;
+
+					// If no pad is available, move near one and wait
+					if (distanceFromHelipad > distanceLength)
```

Comment:
I was thinking of helicopters that are already there for other reasons, because the player ordered them there for example, not because they were sent there for rearming. But I agree that it's a corner case that might not actually matter in the end.